frontend/src/components/grantaccess/GrantAccess.vue

```diff
@@ -216,17 +209,88 @@ function areVerificationsPassed() {
     );
 }
 
-function toSummary() {
-    setGrantAccessFormData(formData.value, forestClientData.value);
-    router.push('/summary');
+function toRequestPayload(
+    formData: any,
+    forestClientNumber: FamForestClient | null
+) {
+    const request = {
+        user_name: formData.userId,
+        user_type_code: formData.domain,
+        role_id: formData.role_id,
+        ...(forestClientNumber
+            ? {
+                  forest_client_number:
+                      forestClientNumber.forest_client_number.padStart(
+                          FOREST_CLIENT_INPUT_MAX_LENGTH,
+                          '0'
+                      ),
+              }
+            : {}),
+    } as FamUserRoleAssignmentCreate;
+    return request;
 }
 
-function roleSelected(evt: any) {
-    let role = applicationRoleOptions.filter((role) => {
-        return role.role_id == evt.value;
-    })[0];
-    grantAccessFormRoleName.value = role.role_name;
-    resetForestClientNumberData();
+async function handleSubmit() {
+    if (forestClientData.value.length > 0) {
+        const successList: string[] = [];
+        const warningList: string[] = [];
+        const errorList: string[] = [];
+        for (const item of forestClientData.value) {
+            const data = toRequestPayload(formData.value, item);
+
+            await userRoleAssignmentApi
+                .createUserRoleAssignment(data as FamUserRoleAssignmentCreate)
+                .then(() => {
+                    successList.push(item.forest_client_number);
+                })
+                .catch((error) => {
+                    if (error.response?.status === 409) {
+                        warningList.push(item.forest_client_number);
+                    } else {
+                        errorList.push(item.forest_client_number);
+                    }
+                });
+        }
+
+        if (successList.length > 0) {
+            pushNotification(
+                Severity.success,
+                `${
+                    formData.value.userId
+                } was successfully added with Client IDs: ${successList.join(
+                    ', '
+                )}`
+            );
+        }
+        if (warningList.length > 0) {
+            pushNotification(
+                Severity.warning,
+                `${
+                    formData.value.userId
+                } already exists with Client IDs: ${warningList.join(', ')}`
+            );
+        }
+        if (errorList.length > 0) {
+            pushNotification(
+                Severity.error,
+                `${
+                    formData.value.userId
+                } was not added with Client IDs: ${errorList.join(', ')}`
+            );
+        }
+
+        router.push('/dashboard');
+    } else {
+        const data = toRequestPayload(formData.value, null);
+        await userRoleAssignmentApi.createUserRoleAssignment(data);
```

Comment:
maybe we should catch the error for this request as well? But I would suggest confirm with @OlgaLiber2 first, how we would like to handle the error case when the role already assigned to the user. For now, we handle them differently for adding a role with forest client number, and without. One uses warning message, one uses toast message. Maybe cut another ticket to confirm with Olga and make them consistent. 

<img width="514" alt="Screen Shot 2023-10-24 at 3 24 45 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/323c5a2a-849e-433b-a497-5edf2fe3e2eb">
<img width="652" alt="Screen Shot 2023-10-24 at 3 26 27 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/a41bdb11-04c8-4dc4-8a05-fb21f89f6ff1">

Reply:
I makes sense to be the same, I'll adjust it.